packages/syft/src/syft/service/network/node_peer.py

```diff
@@ -41,13 +107,66 @@ class NodePeer(SyftObject):
     admin_email: str
 
     def update_routes(self, new_routes: List[NodeRoute]) -> None:
+        print("\n--- inside NodePeer.update_routes ---")
         add_routes = []
-        existing_routes = set(self.node_routes)
+        new_routes: List[NodeRoute] = self.update_route_priorities(new_routes)
         for new_route in new_routes:
-            if new_route not in existing_routes:
+            existed, index = self.existed_route(new_route)
+            if not existed:
                 add_routes.append(new_route)
+            else:
+                print(f"The route {new_route.dict()} already exists!")
+                # if the route already exists, we do not append it to self.new_route,
+                # but update its priority
+                self.node_routes[index].priority = new_route.priority
+
         self.node_routes += add_routes
 
+        print(
+            f"The current highest priority route is {self.pick_highest_priority_route().dict() = }"
+        )
+        print(f"{self = }")
+        print("the current routes in self.node_routes: ")
+        for i, route in enumerate(self.node_routes):
+            print(f"route {i}: {route.dict()}")
+        print()
+
+    def update_route_priorities(self, new_routes: List[NodeRoute]) -> List[NodeRoute]:
+        """
+        Since we pick the newest route has the highest priority, we
+        update the priority of the newly added routes here to be increments of
+        current routes' highest priority.
+        """
+        current_max_priority = max(route.priority for route in self.node_routes)
+        for route in new_routes:
+            route.priority = current_max_priority + 1
+            current_max_priority += 1
+        return new_routes
+
+    def existed_route(self, route: NodeRoute) -> Tuple[bool, int]:
+        """Check if a route exists in self.node_routes
+        - For HTTPNodeRoute: check based on protocol, host_or_ip (url) and port
+        - For PythonNodeRoute: check if the route exists in the set of all node_routes
+        Args:
+            route: the route to be checked
+        Returns:
+            if the route exists, returns (True, index of the existed route in self.node_routes)
+            if the route does not exist returns (False, None)
+        """
+        if isinstance(route, HTTPNodeRoute):
+            for i, r in enumerate(self.node_routes):
+                if (
+                    (route.host_or_ip == r.host_or_ip)
+                    and (route.port == r.port)
+                    and (route.protocol == r.protocol)
+                ):
+                    return (True, i)
+            return (False, None)
+        else:  # PythonNodeRoute
+            if route in set(self.node_routes):
```

Comment:
We might need to do a similar thing here for PythonNodeRoutes ? Since set(self.node_routes) doesn't remove duplicate routes (due to different ids)